crates/sui-graphql-client/src/lib.rs

```diff
@@ -221,7 +221,7 @@ impl Client {
         seq_num: Option<u64>,
     ) -> Result<Option<CheckpointSummary>, Error> {
         ensure!(
-            digest.is_some() != seq_num.is_some(),
+            !(digest.is_some() && seq_num.is_some()),
```

Comment:
||?

Reply:
Both `digest` and `seq_num` can be `None`.

Reply:
i guess the negative logic is tripping me up.

Reply:
I cannot ever figure out how to work with `ensure`. I tried `||`, !(.. || ..)`, and the previous one, they all fail in one case or another.